lib/less/parser/parser.js

```diff
@@ -511,8 +491,24 @@ var Parser = function Parser(context, imports, fileInfo) {
                         }
                         return new(tree.Color)(rgb[1], undefined, '#' + colorCandidateString);
                     }
+                },
 
-                    return this.namedColor();
+                colorKeyword: function () {
```

Comment:
I did not put `colorKeyword()` into `color()` (like it was in colorKeyword) because that way the function is called approximately twice more often (for no real reason).
(Obviously that way it works only because `keyword()` handles named colors in all other contexts, so eventually that will need some refactoring anyway). #2775.